### PR TITLE
update alpaka to 0.3.1 and cupla to 0.1.0

### PR DESCRIPTION
**Pull request by ax3l**
Update alpaka from
 - release  alpaka@0.3.0 (ComputationalRadiationPhysics/alpaka@c0e1b3c56949de248a5151e02282bee9a8cd9463 == [`0.3.0`](https://github.com/ComputationalRadiationPhysics/alpaka/tree/0.3.0) )
to
 - release  alpaka@0.3.1 (ComputationalRadiationPhysics/alpaka@7d3e6ddfb9f8fb834d8895e11aa0d767e39e7b58 == [`0.3.1`](https://github.com/ComputationalRadiationPhysics/alpaka/tree/0.3.1) )

Update cupla from  https://github.com/ComputationalRadiationPhysics/cupla/commit/
 - release  cupla@dev (ComputationalRadiationPhysics/cupla@1cc899ed11a2c7e454a80deb1a2e8e0b06f091d9)
to
 - release  cupla@0.1.0 (ComputationalRadiationPhysics/alpaka@73fb269a8c080e613b0abe860f5d101ba1107e1d == [`0.1.0`](https://github.com/ComputationalRadiationPhysics/cupla/tree/0.1.0) )

### Alpaka

changelog: https://github.com/ComputationalRadiationPhysics/alpaka/releases/tag/0.3.1

7d3e6ddfb Merge pull request #519 from psychocoderHPC/topic-cmakeBackports3
5a0ad99b7 change analysis CUDA test
55b0cb4f9 re-enable header checks for CUDA tests
939d7cc81 add missing inlcudes
a6c737297 test/analysis: fix CMake error
ba8d882ea Tests: fix CMake empty `ALPAKA_CUDA_ARCH`
d03742d80 Add again Boost.Atomic For Fiber
ce9c3bc0d No Boost.Atomic
91338074b Alpaka Include: Link -I
a2a773cf0 Boost: Use CMake Targets
a6f54658c make alpaka a pure interface library
610492c39 Merge pull request #516 from psychocoderHPC/topic-cmakeBackports2
1c9548aa8 CMake: Unify Requirement of 3.7.0+
312948941 CMake: Root Options
32a64f0eb Merge pull request #506 from psychocoderHPC/topic-backPortFixesTo0.3.0
b258071c9 Merge pull request #508 from psychocoderHPC/topic-increaseVersionTo0.3.1
0829bdf7e increase version to 0.3.1
3f652c308 More Undefined ALPAKA_CI defines
9e75fb45e Tests: Fix Undefined Define
1fd1ca0b8 fix getDevCount documentation

### cupla

changelog: https://github.com/ComputationalRadiationPhysics/cupla/blob/0.1.0/CHANGELOG.md

73fb269a8 release version 0.1.0
8dbfc7fd8 Merge pull request #78 from psychocoderHPC/topic-alpakSubmoduleTag0.3.1
268f96b05 update alpaka submodule
7ca202984 Merge pull request #77 from ax3l/doc-changelog010
cd5497cd2 Add RODARE Meta File
b73100f73 Changelog 0.1.0 release
6b40539a0 Merge pull request #76 from psychocoderHPC/fix-copyrightYear
b85672949 fix copyright year
253c297d4 Merge pull request #75 from psychocoderHPC/topic-noLicensePerFileInDoxygen
da8df7cb8 Doxygen: Fix Headers
f3cad02ee Merge pull request #74 from psychocoderHPC/topic-addVersioningFile
ab57a44bb add versioning file
a07bae0d6 Merge pull request #73 from psychocoderHPC/fix-streamAndEventIDCreation
b7b98f8a8 fix id generation for events and streams

## Used Commands

```bash
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="picongpu@hzdr.de" git subtree pull --prefix thirdParty/alpaka git@github.com:ComputationalRadiationPhysics/alpaka.git 0.3.1 --squash

GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="picongpu@hzdr.de" git subtree pull --prefix thirdParty/cupla git@github.com:ComputationalRadiationPhysics/cupla.git 0.1.0 --squash

git rm -rf thirdParty/cupla/alpaka thirdParty/cupla/.gitmodules
git commit
```

### CODE REVIEW

File: thirdParty/cupla/.rodare.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+            "affiliation": "Helmholtz-Zentrum Dresden-Rossendorf",
+            "orcid": "0000-0003-1642-0459"
+        }
+    ],
+    "title": "cupla - C++ User interface for the Platform independent Library Alpaka",
+    "access_right": "open",
+    "upload_type": "software",
+    "license": "LGPL-3.0",
+    "pub_id": "27585",
+    "description": "cupla [qχɑpˈlɑʔ] is a simple user interface for the platform independent parallel kernel acceleration library alpaka. It follows a similar concept as the NVIDIA® CUDA® API by providing a software layer to manage accelerator devices. alpaka is used as backend for cupla."
```
Comment (PrometheusPi) [inline thirdParty/cupla/.rodare.json]: questions:
 - Are these characters `qχɑpˈlɑʔ` correct? 
 - Is a line break possible in json files? (Its quite long.)
 - Should the last sentence start with a a capital letter?

File: thirdParty/cupla/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+0.1.0 "Worf"
+------------
+**Date:** 2018-06-10
+
+Worf: Peace between the Worlds
+
+Following the shining example of Worf, the 0.1.0 release of cupla makes peace
+between the worlds of GPU and CPU computing. The powerful Lords of CUDA
+computing can now speak seamlessly to the inferior worlds of CPUs.
```
Comment (PrometheusPi) [inline thirdParty/cupla/CHANGELOG.md]: questions:
Are you serious?  😁

File: thirdParty/cupla/example/CUDASamples/matrixMul/src/matrixMul.cpp [hunk 3]
```diff
@@ -1,5 +1,4 @@
-/**
- * Copyright 1993-2014 NVIDIA Corporation.  All rights reserved.
+/* Copyright 1993-2014 NVIDIA Corporation.  All rights reserved.
```
Comment (PrometheusPi) [inline thirdParty/cupla/example/CUDASamples/matrixMul/src/matrixMul.cpp]: Is this Okay regarding the license?

File: thirdParty/cupla/example/CUDASamples/vectorAdd/src/vectorAdd.cpp [hunk 4]
```diff
@@ -1,5 +1,4 @@
-/**
- * Copyright 1993-2015 NVIDIA Corporation.  All rights reserved.
+/* Copyright 1993-2015 NVIDIA Corporation.  All rights reserved.
```
Comment (PrometheusPi) [inline thirdParty/cupla/example/CUDASamples/vectorAdd/src/vectorAdd.cpp]: Same here

File: thirdParty/cupla/.rodare.json (hunk 1 above)
Comment (ax3l) [reply thirdParty/cupla/.rodare.json]: first two yes and the last one is a name and we decided to spell alpaka lowercase.

File: thirdParty/cupla/CHANGELOG.md (hunk 2 above)
Comment (ax3l) [reply thirdParty/cupla/CHANGELOG.md]: that's exactly what cupla does :D

File: thirdParty/cupla/example/CUDASamples/matrixMul/src/matrixMul.cpp (hunk 3 above)
Comment (ax3l) [reply thirdParty/cupla/example/CUDASamples/matrixMul/src/matrixMul.cpp]: this is *cupla* source code. there is not reason to review it. here

File: thirdParty/cupla/example/CUDASamples/vectorAdd/src/vectorAdd.cpp (hunk 4 above)
Comment (ax3l) [reply thirdParty/cupla/example/CUDASamples/vectorAdd/src/vectorAdd.cpp]: same as above

**Pull request merged**